### DISCUSSION TITLE
opinion in the website is missing in API

### DISCUSSION THREAD

**Discussion by raminass** [freelawproject/courtlistener#3464] [General]
Hey,

opinion: https://www.courtlistener.com/opinion/4403794/pavan-v-smith/

tried to located in api using Docket Number: 16–992 as appears in the website using the below request:
https://www.courtlistener.com/api/rest/v3/dockets/?docket_number=16-992

However the results are for different cases. 

Also, what's the best way to locate opinions in the api given the web url of the opinion ?

**Comment by mlissner** [comment]
This is a messy data problem, I'm afraid. When you use the docket endpoint it's not a fuzzy search, it's an exact match. 

In this case, the URL you should have been using is:

https://www.courtlistener.com/api/rest/v3/dockets/?docket_number=16%E2%80%93992.

The difference is that this case had an em dash in the docket number, and a period at the end. I just cleaned it up though, so your original query now works.

That's good, but using the API probably isn't the best way to do this, for the reasons mentioned.

If you're looking up something from a web URL, I'd just use the cluster API:

https://www.courtlistener.com/api/rest/v3/clusters/4403794/

I've made an issue to clean up em-dashes here: https://github.com/freelawproject/courtlistener/issues/3465

**Comment by raminass** [comment]
Thanks for the reply, another issue now with same case, I see a discrepancy in data same opinion: https://www.courtlistener.com/api/rest/v3/opinions/4181047/

Returns:
    "author_str": "Per Curiam",
    "per_curiam": false,

which one is trusted ?

**Reply by mlissner** [reply]
Yeah, I'd go with a combination of the two, I guess. Obviously, we haven't updated the boolean field properly, so I opened another issue to get back around to doing that. But if it says Per Curiam in the author_str, you can trust that, and if the `per_curiam` field is true, you can trust that too. Here's the latest issue: 

https://github.com/freelawproject/courtlistener/issues/3466

Thanks for identifying these.

**Comment by raminass** [comment]
Thanks again, so how would I get per curiam opinions/clusters using the author_str in the api, that filed after particular date.

**Reply by mlissner** [reply]
There should be some good documentation about filtering by date here or in the documentation itself. If you're having trouble with that, let me know and I can find a reference or answer here again. 

You're right that you can't filter by the `author_str`, so, yeah, this does become a bit of an issue. I guess the way to fix this is to:

1. We fix it in our data. This isn't far-fetched; it should be pretty easy, I think, and we're often fixing things.

3. You could use the bulk data and filter that way.

Sorry! You're finding some good bugs, and the workarounds are getting trickier. Let us know how you want to proceed, and if you prefer that we do the data fix, some idea of your timeline and urgency would be helpful too.
